### PR TITLE
[Merged by Bors] - fix: improvements I noticed when teaching

### PR DESCRIPTION
**Pull request by fpvandoorn**
* Rename (and generalize and move) 

```
Int.units_ne_neg_self -> units_ne_neg_self
Int.neg_units_ne_self -> neg_units_ne_self
```
* Change the `simps` config for `Closeds` 
* Add some `gcongr`-lemmas (currently is a bit picky about the exact statement of lemmas tagged with `gcongr`, so I had to add some variants that I could tag).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Sets/Closeds.lean [hunk 1]
```diff
@@ -105,7 +106,7 @@ instance : Inhabited (Closeds α) :=
   ⟨⊥⟩
 
 @[simp, norm_cast]
-theorem coe_sup (s t : Closeds α) : (↑(s ⊔ t) : Set α) = ↑s ∪ ↑t :=
+theorem coe_sup (s t : Closeds α) : (↑(s ⊔ t) : Set α) = ↑s ∪ ↑t := by
```
Comment (sgouezel) [inline Mathlib/Topology/Sets/Closeds.lean:109]: why did you change this one?

**Review by sgouezel — COMMENTED** [review]
bors d+

File: Mathlib/Topology/Sets/Closeds.lean (hunk 1 above)
Comment (fpvandoorn) [reply Mathlib/Topology/Sets/Closeds.lean:109]: I can revert this, but I needed to see the statement in the infoview to click on the types of things to ensure that the statement said the correct thing. It is annoying that this is only possible in tactic mode.

**Comment by fpvandoorn** [comment]
bors merge
